src/transformers/integrations.py

```diff
@@ -559,20 +558,12 @@ def __init__(self):
         if has_wandb:
             import wandb
 
-            wandb.ensure_configured()
-            if wandb.api.api_key is None:
-                has_wandb = False
-                logger.warning(
-                    "W&B installed but not logged in. Run `wandb login` or set the WANDB_API_KEY env variable."
-                )
-                self._wandb = None
-            else:
-                self._wandb = wandb
```

Comment:
Why is it not necessary anymore?

Reply:
They entire login logic is now directly handled by wandb. If not logged in:
* In an interactive terminal it will prompt for the key (first time only)
* In a non interactive terminal it will log in offline mode (meaning the user can later sync the runs through a command)